### DISCUSSION TITLE
fuzzing multiple targets round-robin

### DISCUSSION THREAD

**Discussion by martinus** [AFLplusplus/AFLplusplus#2210] [Q&A]
I've currently a setup where I'd like to continuously fuzz about 20 targets or so (will likely increase over time)... Currently I basically have a script that has an infinite loop over all targets and fuzzes each with `AFL_EXIT_ON_TIME` set to ~10 minutes. So when nothing is found after 10 minutes, the next target is fuzzed.  This seems to work well as it gives targets where more is found more time too.

What I'm wondering though, is that advisable to stop & start the fuzzers so often? startup seems to be fast for my use cases (takes a few seconds at most), but I notice the "pending" number is always quite high that way.

I have also set `AFL_AUTORESUME` and `AFL_FINAL_SYNC`.

**Comment by domenukk** [comment]
Historically the performance to resume used to be really bad - the fuzzer would have spent a considerable amount of the 10 minutes just going through the original queue.
However, AFL++ recently added the fast resume option and with that this should be fine, I _think_?
Basically it will load the state from a file instead of rebuilding it on startup.

**Comment by vanhauser-thc** [comment]
if you run the most current stable branch then there is the fast resume feature (automatically active) - however this only works if the target is not recompiled.
Generally what you are doing does not sound very smart to be honest. It would be more advisable to fuzz them for longer, 10 minutes is nothing.

**Reply by martinus** [reply]
I guess my real question is, is it (much) worse to fuzz a total of 1 day (with stop & start every 10 minutes) than fuzzing continuously for a day? Startup seems quite fast for me.
